plugins/woocommerce/includes/wc-core-functions.php

```diff
@@ -2583,3 +2583,30 @@ function wc_cache_get_multiple( $keys, $group = '', $force = false ) {
 	}
 	return $values;
 }
+
+/**
+ * Enable WebP images in WooCommerce.
+ *
+ * WebP files should be able to be imported in WooCommerce, as they've been supported in WordPress core since v5.8.
+ */
+add_filter( 'woocommerce_rest_allowed_image_mime_types', 'more_mimes_to_exts' );
+function more_mimes_to_exts( $mime_to_ext ){
+	$mime_to_ext['webp'] = 'image/webp';
+
+	return $mime_to_ext;
+}
+
+function wc_rest_allowed_image_mime_types() {
```

Comment:
This function already exists: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-rest-functions.php#L47:L59

Adding a second definition will cause a fatal error. You need to edit the original function.

Reply:
As I understand, I should go ahead with removing the whole function that starts at line 2599, at wc-core-functions.php, and add `'webp'         => 'image/webp',` in line 57 at wc-rest-functions.php.

Correct?

Reply:
I went ahead with doing so.